include/fc/crypto/sha256.hpp

```diff
@@ -3,6 +3,7 @@
 #include <fc/string.hpp>
 #include <fc/platform_independence.hpp>
 #include <fc/io/raw_fwd.hpp>
+#include <boost/functional/hash.hpp>
```

Comment:
AFAICS boost::hash<fc::sha256> isn't used anywhere, so you should be able to remove the include and the template specialization below.
I'm not particularly happy with it (and std::hash<...>) because the return value is architecture-dependent. I don't think this is used in a consensus-related context though, so not that bad.

Reply:
Perhaps off-topic here, perhaps `std::hash` is used in `hashed_unique` indexes?

Reply:
At first, I actually commented out the sha256 specialization that needed this header, and it compiled and ran fine. With the other changes being fairly easy, this was the change that I questioned most. So instead of commenting it out, I put it back in and added the header. I will do more research to see if I can find anywhere that it is being used.

Reply:
> Perhaps off-topic here, perhaps std::hash is used in hashed_unique indexes?

Yes, std::hash is but boost::hash isn't.

Reply:
I removed boost::hash<fc::sha256> from the file. Everything compiles and runs fine without it. Please take a look at the latest commit.